src/sortStudents.ts

```diff
@@ -1,16 +1,82 @@
-
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
-// create SortOrder type
-export type SortOrder;
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: string,
+): Student[] {
+  let sortedStudents: Student[] = [...students];
+
+  switch (sortBy) {
+    case SortType.Name:
+    case SortType.Surname:
+      if (order === 'asc') {
+        sortedStudents = sortedStudents.sort(
+          (firstStudent, nextStudent) => firstStudent[sortBy]
+            .localeCompare(nextStudent[sortBy]),
+        );
+      } else {
+        sortedStudents = sortedStudents.sort(
+          (firstStudent, nextStudent) => nextStudent[sortBy]
+            .localeCompare(firstStudent[sortBy]),
+        );
+      }
+      break;
```

Comment:
Can be shortened, same for the rest of cases
```suggestion
  const sortOrder = order === 'asc' ? 1 : -1;
  switch (sortBy) {
    case SortType.Name:
    case SortType.Surname:
      sortedStudents = sortedStudents.sort(
        (firstStudent, nextStudent) => sortOrder * firstStudent[sortBy]
          .localeCompare(nextStudent[sortBy]),
      );
      break;
```